plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt

```diff
@@ -64,14 +63,17 @@ internal class JKCodeBuilder(context: NewJ2kConverterContext) {
         }
 
         private fun renderModifiersList(modifiersListOwner: JKModifiersListOwner) {
-            val isOpenByDefault = (modifiersListOwner is JKClass && modifiersListOwner.classKind == INTERFACE) ||
-                    (modifiersListOwner is JKMethod && modifiersListOwner.parentOfType<JKClass>()?.classKind == INTERFACE)
+            var isOpenAndAbstractByDefault = false
+            if (modifiersListOwner is JKElement) {
```

Comment:
If you need a smart cast to a `JKElement` to have the code work both for `JKMethod` and `JKField`, maybe you could preserve the structure of the existing code and just change `modifiersListOwner is JKMethod` to `modifiersListOwner is JKDeclaration`?

Reply:
made the change to `JKDeclaration`!